src/v/storage/disk_log_impl.cc

```diff
@@ -534,8 +534,13 @@ ss::future<compaction_result> disk_log_impl::compact_adjacent_segments(
             throw std::runtime_error(fmt::format(
               "Aborting compaction of closed segment: {}", *segment));
         }
+        // find max generation of compacted segments
+        result_generation_id = std::max(result_generation_id, *gen_it);
```

Comment:
Any particular reason why we do a max instead of just taking the generation of the first segment? I think we can detect truncation of other